encodings/fastlanes/src/bitpacking/serde.rs

```diff
@@ -66,9 +71,13 @@ impl SerdeVTable<BitPackedVTable> for BitPackedVTable {
             }
         };
 
-        // Load validity from the zero'th or second child, depending on whether patches are present.
-        let validity = if metadata.patches.is_some() {
-            load_validity(2)?
+        let validity = if let Some(patches_meta) = &metadata.patches {
+            let validity_idx = if patches_meta.chunk_offsets_dtype().is_some() {
+                3
+            } else {
+                2
+            };
+            load_validity(validity_idx)?
         } else {
             load_validity(0)?
         };
```

Comment:
can we clean this up a little
```suggestion
        let validity_idx = if let Some(patches_meta) = &metadata.patches {
            if patches_meta.chunk_offsets_dtype().is_some() {
                3
            } else {
                2
            }
} else {
0
};
let validity = load_validity(validity_idx)?
```

Reply:
done